src/plugins/path/README.md

```diff
@@ -5,28 +5,88 @@
 - infos/needs =
 - infos/provides = check
 - infos/status = maintained nodep libc nodoc
-- infos/metadata = check/path
-- infos/description = Checks if keys enriched with appropriate metadata contain valid paths as values
+- infos/metadata = check/path check/permission/types check/permission/user
+- infos/description = Checks if keys enriched with appropriate metadata contain valid paths as values as well
+as correct permissions
 
 ## Introduction
 
-This plugin checks whether the value of a key is a valid file system path. 
+This plugin checks whether the value of a key is a valid file system path and optionally if
+correct permissions are set for a certain user.
 
 ## Purpose
 
 The motivation to write this plugin is given by the two paths that exist
-in /etc/fstab: the device file and the mount point. A missing file is
+in /etc/fstab: the device file and the mountpoint. A missing file is
 not necessarily an error, because the device file may appear later when
-a device is plugged in and the mount point may be there when another
+a device is plugged in and the mountpoint may be there when another
 subsequent mount was executed. So only warnings are yielded in that
 case. One situation, however, presents an error: Only an absolute path
-is allowed to occur for both device and mount point. When checking for
+is allowed to occur for both device and mountpoint. When checking for
 relative files, it is not enough to look at the first character if it is
 a `/`, because remote file systems and some special names are valid, too.
 
+If `check/permission/types = [permission]` is also present it will check for the correct permissions
+of the file/directory. Optionally, you can also add `check/permission/user = [user]"` which then checks the permissions
+for the given user. When calling `kdb set` on the actual key, you have to run as `root` user
+or the file permissions cannot be checked (you will receive an error message).
+
+ `check/permission/types = rw` and `check/permission/user = tomcat` for example will check if the user
```

Comment:
How to do a shellrecorder syntax example when it is dependent on the user who is executing the commands. 

The example above is actually fully functional and I use it to copy paste it and try out if the intended plugin functionality works (not user tomcat but my own user though)

Are those tests done as `root`?

Reply:
> How to do a shellrecorder syntax example when it is dependent on the user who is executing the commands.

There may be a better solution but the simplest way would be to only check exit codes and not the actual messages. Alternatively you can use a regex that simply ignores user specific data for the messages.

> Are those tests done as root?

Using `sudo` should work. (used [here](https://github.com/ElektraInitiative/libelektra/tree/master/src/plugins/boolean#example))

Reply:
> Using `sudo` should work.

Unfortunately I do not think that will work. Currently the Markdown Shell Recorder [just strips `sudo` from commands](https://github.com/ElektraInitiative/libelektra/blob/master/tests/shell/shell_recorder/tutorial_wrapper/markdown_shell_recorder.sh#L29) before it executes them.

Reply:
Ok, since a full valid example not only requires sudo but also is dependent on the system structure as well as current user, a shellrecorder tested example seems not feasible

Reply:
The plugin should do something useful also if it is not executed as root. And at least that can be tested in shell recorder.

Reply:
I changed the example now, I am not sure if this is reproducable on all systems. 
Is this something you expected?